authentik/policies/expression/evaluator.py

```diff
@@ -44,13 +44,14 @@ def expr_func_user_has_authenticator(
     ) -> bool:
         """Check if a user has any authenticator devices, optionally matching *device_type*"""
         user_devices = devices_for_user(user)
-        if device_type:
-            for device in user_devices:
+        for device in user_devices:
+            if device_type:
                 device_class = device.__class__.__name__.lower().replace("device", "")
                 if device_class == device_type:
                     return True
-            return False
-        return len(user_devices) > 0
+            else:
```

Comment:
using `len(list())` would also make this a little bit simpler, without the additional else

Reply:
Sure, whichever way you'd prefer it.  len(list()) is definitely simpler, but has to create the entire list instead of just the first item.  It's not like the user's authenticator list would ever be massive though.  Updated to the simpler version.